sdk/core/core-lro/src/poller.ts

```diff
@@ -0,0 +1,104 @@
+import { PollOperation } from "./pollOperation";
+
+type CancelOnProgress = () => void;
+
+export abstract class Poller<TProperties, TResult> {
+  private stopped: boolean;
+  private resolve?: (value: TResult) => void;
+  private reject?: (error: Error) => void;
+  private pollOncePromise?: Promise<void>;
+  private promise: Promise<TResult>;
+  private onProgressCallback?: (op: PollOperation<TProperties>) => void;
+  public operation: PollOperation<TProperties>;
+
+  constructor(operation: PollOperation<TProperties>, stopped: boolean = false) {
+    this.operation = operation;
+    this.stopped = stopped;
+    this.promise = new Promise((resolve, reject) => {
+      this.resolve = resolve;
+      this.reject = reject;
+    });
+    this.operation.state.started = true;
+    if (!this.stopped) {
```

Comment:
we shouldn't poll if `this.isDone()` either, right? Or since this check already exists inside `startPolling()`, maybe we don't need it?